.travis.yml

```diff
@@ -0,0 +1,37 @@
+language: python
+
+python:
+  - "3.6"
+
+# The tests run only if it is an PR or if a tag is pushed. This should prevent duplicate
+# builds with pr and push. We cannot disable push as it also disables deployment via
+# tags.
+if: type != push OR tag IS present OR branch = master
+
+install:
+  - sudo apt-get update
+  - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - PATH="$HOME/miniconda/bin:$PATH"
+  - conda env create -f configurations/environment.yml
+  - source activate structExpectations-analysis
+
+before_script:
+  - export PYTHONPATH=$PYTHONPATH:$(pwd)
+  - export PYTHONPATH=$PYTHONPATH:$(pwd)/submodules/soepy
+
+script:
+    # Test code quality
+    - flake8
+    - black .
+
+    # pytest submodules
+    - py.test
+
+    # Test estimation run
+    - cd python; python create_report.py --all; cd ..
```

Comment:
Please set up a link `create-report` and allow for it so be directly executed by moving it in `bin` as we do for `update-repository`.

Call `create-report` directly in `basecamp`.